tests/src/cli.js

```diff
@@ -4,82 +4,144 @@
 import path from 'path';
 
 import { expect } from 'chai';
-import { CLIEngine } from 'eslint';
+import { CLIEngine, ESLint } from 'eslint';
 import eslintPkg from 'eslint/package.json';
 import semver from 'semver';
+import * as importPlugin from '../../src/index';
 
 describe('CLI regression tests', function () {
   describe('issue #210', function () {
+    let eslint;
     let cli;
     before(function () {
-      cli = new CLIEngine({
-        useEslintrc: false,
-        configFile: './tests/files/issue210.config.js',
-        rulePaths: ['./src/rules'],
-        rules: {
-          'named': 2,
-        },
-      });
+      if(ESLint) {
+        eslint = new ESLint({
+          useEslintrc: false,
+          overrideConfigFile: './tests/files/issue210.config.js',
+          rulePaths: ['./src/rules'],
+          overrideConfig: {
+            rules: {
+              'named': 2,
+            },
+          },
+          plugins: { 'eslint-plugin-import': importPlugin },
+        });
+      } else {
+        cli = new CLIEngine({
+          useEslintrc: false,
+          configFile: './tests/files/issue210.config.js',
+          rulePaths: ['./src/rules'],
+          rules: {
+            'named': 2,
+          },
+        });
+        cli.addPlugin('eslint-plugin-import', importPlugin);
+      }
     });
     it("doesn't throw an error on gratuitous, erroneous self-reference", function () {
-      expect(() => cli.executeOnFiles(['./tests/files/issue210.js'])).not.to.throw();
+      if (eslint) {
+        return eslint.lintFiles(['./tests/files/issue210.js'])
+          .then(() => { }, () => expect.fail() );
```

Comment:
any assertion for the happy path here? if not:
```suggestion
          .catch() => expect.fail());
```
or actually, you could just return the promise, and mocha will fail if it rejects.

Reply:
I didn't know that. Thank you for letting me know!